### ISSUE TITLE
Agent's size

### ISSUE THREAD

**Issue by hcv1027** [allenai/ai2thor#266]
Can I change the agent's size? For example, set the agent's width as 45cm, height as 42cm in horizontal view. This agent's size would change the collision range.

**Comment by hcv1027** [comment]
Is the code `m_CharacterController.radius = 0.2f;` in `ai2thor/unity/Assets/Scripts/BaseFPSAgentController.cs` what I want?

**Comment by winthos** [comment]
yes, you can adjust the radius and height of the agent by editing that line. If you edit `m_CharacterController.height` you can also change the height from the default `1.8`.

Be warned that you might see some inconsistent behavior if you change the width or height too much, as the starting position of the Agent in scenes are such that the radius and height are assumed to be 0.2m and 1.8m respectively. For example, setting the radius to 1m might cause the agent to spawn into a scene and clip through the wall, or become stuck in other geometry.

**Comment by hcv1027** [comment]
Hi @winthos , is it possible to avoid the problem you mentioned by using `TeleportFull` action to set the agent's position, where the positions are got from the action of `GetReachablePositions`?
```python
event = controller.step(dict(action='Initialize'))
event = controller.step(dict(action='GetReachablePositions'))
reachable_positions = event.metadata['reachablePositions']
position = reachable_positions[random.randrange(len(reachable_positions))]
rotation = random.uniform(0.0, 360.0)
event = self.controller.step(dict(action='TeleportFull',
    x=position['x'], y=position['y'], z=position['z'],
    rotation=rotation, horizon=30.0))
```

**Comment by winthos** [comment]
That may help catch some cases, but it will not be able to help in cases where the initial position of the agent is already clipping into the environment. `GetReachablePositions` calculates the reachable positions based on the starting position of the agent, and if that starting position is itself already bad it will not produce reliable positions.
